app/src/main/java/ch/epfl/skysync/models/flight/PlannedFlight.kt

```diff
@@ -15,6 +16,44 @@ data class PlannedFlight(
     override val vehicles: List<Vehicle>
 ) : Flight {
   fun readyToBeConfirmed(): Boolean {
-    return team.isComplete() && nPassengers > 0 && balloon != null && basket != null
+    return team.isComplete() &&
+            nPassengers > 0 &&
+            balloon != null &&
+            basket != null &&
+            vehicles.isNotEmpty()
   }
+
+    fun confirmFlight(
+        notReadyToConfirm: ()->Unit,
+        meetupTimeTeam: LocalTime,
+        departureTimeTeam: LocalTime,
+        meetupTimePassenger: LocalTime,
+        meetupLocationPassenger: String,
+        remarks: List<String>,
+        color: FlightColor
+
+
+    ): ConfirmedFlight {
+        if (!readyToBeConfirmed()) {
+            notReadyToConfirm()
```

Comment:
I'm not convinced by the `notReadyToConfirm` callback, instead just call `readyToBeConfirmed` before:
```kotlin
if (flight.readyToBeConfirmed()) {
    val confirmedFlight = flight.confirmFlight()
}
```
And throw an exception if `confirmFlight` is called on an not ready to be confirmed flight.
(Also, at the moment you don't return after executing the callback, so you still return a confirmed flight even if it is not ready to)

Reply:
yep, thanks, you were too quick:-) I've replaced it with an exception